api/src/main/java/jakarta/data/exceptions/FunctionalityNotSupportedException.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.exceptions;
+
+/**
+ * This exception is raised when the data store is not capable of performing a particular function.
+ * For example, some NoSQL databases are not capable of counting a total number of results,
+ * and therefore raise this exception from repository methods that are declared to return
+ * {@link jakarta.data.repository.Page Page}, which requires the total number of results.
+ */
+public class FunctionalityNotSupportedException extends MappingException {
+    private static final long serialVersionUID = 4093503320173860847L;
+
+    /**
+     * Constructs a new FunctionalityNotSupportedException exception with the specified detail message.
+     *
+     * @param message the detail message.
+     */
+    public FunctionalityNotSupportedException(String message) {
```

Comment:
why not use the [UnsupportedOperationException](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/UnsupportedOperationException.html)?

Reply:
Good idea. No one on the call had thought of that. I agree, it's best to reuse a standard exception that fits.  It is updated now in [this commit](https://github.com/jakartaee/data/pull/146/commits/a700142e37878b94d0981ab3a383e3f60046f09d), please review again.